src/terminal/input/terminalInput.cpp

```diff
@@ -504,7 +135,35 @@ TerminalInput::OutputType TerminalInput::HandleKey(const INPUT_RECORD& event)
         {
             _lastVirtualKeyCode = std::nullopt;
         }
-        return MakeUnhandled();
+        // If NumLock is on, and this is an Alt release with a unicode char,
+        // it must be the generated character from an Alt-Numpad composition.
+        if (WI_IsFlagSet(controlKeyState, NUMLOCK_ON) && virtualKeyCode == VK_MENU && unicodeChar != 0)

```

Comment:
I wonder - will Alt+Numpad ever produce a leading/trailing surrogate pair?

Reply:
If I remember correctly, Alt+Numpad is just for codepoints 0 to 255, and there's another variant (involving the plus key maybe?) which is supposed to handle Unicode characters. That's not currently supported.